src/elo/util/dbInterface.ts

```diff
@@ -167,11 +180,25 @@ export const writeEloSnapshots = async (
   snapshots: EloSnapshotIM[],
   context: GraphQLContext
 ) => {
-  return context.prisma.eloSnapshot.createMany({
-    data: snapshots,
-  });
+  const upserts = snapshots.map(snapshot =>
+    context.prisma.eloSnapshot.upsert({
+      where: {
+        playerId_matchId_versionId: {
+          playerId: snapshot.playerId,
+          matchId: snapshot.matchId,
+          versionId: snapshot.versionId,
+        },
+      },
+      // do not update existing snapshot
+      update: { },
+      create: snapshot,
+    })
+  );
+
+  return Promise.all(upserts);
```

Comment:
Good approach to resolving the promises created here. I wonder how this will affect performance with all the upserts, especially for larger scale recalculations on all or most of the database, but if it becomes prohibitively inefficient we can put more concerted effort in in the future.